proposals/0000-a11y-actions.md

```diff
@@ -0,0 +1,84 @@
+---title: Accessibility Actions Support
+author:
+- Marc Mulcahy
+date: 2018-10-31
+---
+
+# RFC0000: Accessibility Actions Support
+
+## Summary
+
+Custom components often require interactions beyond simple touches. Programatic access to these interactions must be provided to allow disabled users to perform them.
+
+## Basic example
+
+Consider the simple example of a custom slider. The newly expanded accessibilityRole property allows us to indicate that the component is a slider in a cross-platform way. However, the blind or motor impaired user can't adjust the slider, since a gesture other than a single tap is required to do so.
+
+## Motivation
+
+The example above is one use case, but there are several common patterns and use cases which require custom action support. Here are a few examples:
+
+* Adjustable controls (such as sliders)
+* Scrollable containers where the scrolling is not handled natively
+* Extra swipe actions in a list (swipe right to delete an email message for example)
+
+## Detailed design
+
+This proposal adds an additional accessibility-related property called accessibilityActions to View. The value of this property is a list of accessibility action objects. Each action contains the following properties:
+
+* name: a string name for the action. Note that a pre-defined set of standard actions is supported, but components can also define their own custom actions.
+* description: a localized description of the action which can be presented to the user to describe what will happen when the action is performed.
+* function: the function to call when the assistive technology requests this action to be performed. This function should return true if the action is performed, and false if not.
+
+  Standard action names include:
+
+* activate: activate the item (typically accomplished by double tapping the item when using a screen reader)
+* longpress: Long presses anywhere inside the component
+* scrollforward: Scrolls the component forward one screen full
+* scrollbackward: Scrolls the component backward one screen full
+* dismiss: Dismisses the component
+* magictap: on iOS, user double taps with two fingers while accessibility focus is on the component. Not available on Android.
+* adjustup: Adjusts component up
+* adjustdown: adjust component down
+* expand: Expands the component
+* collapse: Collapses the component
+* copy: copies content in the component to the clipboard
+* cut: cuts content in the component to the clipboard
+* paste: pastes clipboard contents into this component
+* setvalue: sets the value of this component (typically used to adjust sliders, etc.)
+
+### Implementation Notes
```

Comment:
What's the proposed story for web? I don't think there are any equivalent accessibility APIs

Reply:
I don't think so either. But I think we'll need action support for native, especially given the behavior on Android 8.0 and above, where the framework no longer simulates a touch event to the center of a component's bounding box, but instead generates a click accessibility action. Most standard Android views handle this, but some React native views like InputText don't. Probably also applies to some handling in iOS, where clicking on the center of a bounding box doesn't make sense, like links which span multiple lines.

Reply:
@necolas I can't come up with a scenario where we'd need the equivalent accessibility API on web, since we don't have a source of actions from an AX service like Native platforms have.